lib/internal/http2/core.js

```diff
@@ -441,6 +441,8 @@ function afterDoStreamWrite(status, handle, req) {
 
 function onHandleFinish() {
   var session = this[kSession];
+  if (!session)
```

Comment:
Yep, this was an area that I've been planning to get back to (specifically, the sequencing of the events and a well defined event lifecycle)